pyowc/calibration.py

```diff
@@ -540,22 +531,160 @@ def update_salinity_mapping(float_dir, float_name, config):
                     float_name + config['FLOAT_MAPPED_POSTFIX']
 
     # save the data
-    scipy.savemat(save_location, {'la_ptmp': la_ptmp,
-                                  'la_mapped_sal': la_mapped_sal,
-                                  'la_mapsalerrors': la_mapsalerrors,
-                                  'scale_long_large': scale_long_large,
-                                  'scale_lat_large': scale_lat_large,
-                                  'scale_long_small': scale_long_small,
-                                  'scale_lat_small': scale_lat_small,
-                                  'scale_phi_large': scale_phi_large,
-                                  'scale_phi_small': scale_phi_small,
-                                  'scale_age_large': scale_age_large,
-                                  'scale_age_small': scale_age_small,
-                                  'use_pv': use_pv,
-                                  'use_saf': use_saf,
-                                  'p_delta': p_delta,
-                                  'p_exclude': p_exclude,
-                                  'la_noise_sal': la_noise_sal,
-                                  'la_signal_sal': la_signal_sal,
-                                  'la_profile_no': la_profile_no,
-                                  'selected_hist': selected_hist})
+    savemat(save_location, {'la_ptmp': la_ptmp,
+                            'la_mapped_sal': la_mapped_sal,
+                            'la_mapsalerrors': la_mapsalerrors,
+                            'scale_long_large': scale_long_large,
+                            'scale_lat_large': scale_lat_large,
+                            'scale_long_small': scale_long_small,
+                            'scale_lat_small': scale_lat_small,
+                            'scale_phi_large': scale_phi_large,
+                            'scale_phi_small': scale_phi_small,
+                            'scale_age_large': scale_age_large,
+                            'scale_age_small': scale_age_small,
+                            'use_pv': use_pv,
+                            'use_saf': use_saf,
+                            'p_delta': p_delta,
+                            'p_exclude': p_exclude,
+                            'la_noise_sal': la_noise_sal,
+                            'la_signal_sal': la_signal_sal,
+                            'la_profile_no': la_profile_no,
+                            'selected_hist': selected_hist})
+
+
+#pylint: disable=too-many-locals
+#pylint: disable=too-many-statements
+#pylint: disable=too-many-arguments
+def set_calseries(float_dir, float_name, system_config):
```

Comment:
I realise the logic of putting these two functions here, but the update_salinity_mapping routine does not actually do any calibrating, it just creates a huge grid of expected values for salinity on the same pressure levels as the float salinity. We then use this grid in the calculate_piecewisefit to calibrate the float that we are analysising.

Setting the calseries could be seen as a configuration too - perhaps it would be more at home in the configuration folder, and calculate piecewise fit would be more at home here?

Reply:
To add to this:

- Should update salinity mapping be changed to something like calc_salinity_map
- Should calculate piecewise fit be changed to something like calc_calibration

To make it more obvious what each function does?

Reply:
since update_salinity_mapping was the facade of the process up to now, I put it here, because I foresee the calibration.py to be the main facade 
but creating the grid of expected values is part of the calibration process rather than configuration, no ?

Reply:
> To add to this:
> 
>     * Should update salinity mapping be changed to something like calc_salinity_map
> 
>     * Should calculate piecewise fit be changed to something like calc_calibration
> 
> 
> To make it more obvious what each function does?

I agree that (many) functions could be renamed to better reflect what they do, but I would keep renaming for the next refactoring step 2 to avoid bringing to much changes at one time.
this also will help to keep discussing with the Matlab routines experts when we'll consider optimization and function content refactoring .

Reply:
Yes, apologies, I didn't make myself clear.

- The salinity mapping is part of the calibration process, or rather the output is. The mapped grid is used as a kind of control for the calibration.
- The section I was referring to moving was set_calseries. This file contains the set up parameters for calculating the calibration, and therefore would probably be better in the configuration folder?

Reply:
I'm not yet familiar enough with the actual work of each functions, you're surely right about this.
I see that it takes the configuration as input and produce a file, so yes I'll move set_calseries to configuration.py